modules/account-lib/src/coin/sol/stakingActivateBuilder.ts

```diff
@@ -0,0 +1,109 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { TransactionType } from '../baseCoin';
+import { Transaction } from './transaction';
+import { TransactionBuilder } from './transactionBuilder';
+import { InstructionBuilderTypes } from './constants';
+
+import assert from 'assert';
+import { StakingActivate } from './iface';
+import { isValidPublicKey, isValidStakingAmount } from './utils';
+import { BuildTransactionError } from '../baseCoin/errors';
+
+export class StakingActivateBuilder extends TransactionBuilder {
+  protected _amount: string;
+  protected _stakingAddress: string;
+  protected _validator: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.StakingActivate;
+  }
+
+  /** @inheritdoc */
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    for (const instruction of this._instructionsData) {
+      if (instruction.type === InstructionBuilderTypes.StakingActivate) {
+        const activateInstruction: StakingActivate = instruction;
+        this.sender(activateInstruction.params.fromAddress);
+        this.stakingAddress(activateInstruction.params.stakingAddress);
+        this.amount(activateInstruction.params.amount);
+        this.validator(activateInstruction.params.validator);
+      }
+    }
+  }
+
+  /**
+   * The amount to stake expressed in Lamports, 1 SOL = 1_000_000_000 lamports
+   *
+   * @param {string} amount expressed in Lamports
+   * @returns {StakeBuilder} This staking builder.
+   *
+   */
+  amount(amount: string): this {
+    if (!isValidStakingAmount(amount)) {
+      throw new BuildTransactionError('Value cannot be zero or less');
+    }
+    this._amount = amount;
+    return this;
+  }
+
+  /**
+   * The staking address of the staking account.
+   *
+   * @param {string} stakingAddress public address of the staking account
+   * @returns {StakeBuilder} This staking builder.
+   *
+   * @see https://docs.solana.com/staking/stake-accounts#account-address
+   */
+  stakingAddress(stakingAddress: string): this {
+    if (!stakingAddress || !isValidPublicKey(stakingAddress)) {
+      throw new BuildTransactionError('Invalid or missing stakingAddress, got: ' + stakingAddress);
+    }
+    this._stakingAddress = stakingAddress;
+    return this;
+  }
+
+  /**
+   * Set validator address to vote
+   *
+   * @param {string} validator address of the validator to vote
```

Comment:
same here: "Validator address to delegate funds to"

Reply:
Ty